### DISCUSSION TITLE
Need help setting up SDK (Macbook Sonoma)

### DISCUSSION THREAD

**Discussion by jtgladden** [ps2dev/ps2sdk#702] [Q&A] [answered]
Hi! I have been interested in compiling code that would run on a PS2 for a few years now, but I had previously given up. The past few days my dreams have been rekindled and I've found [this article](https://glampert.com/2015/02-27/ps2-homebrew-setting-up-the-environment/) that explains how to install the SDK on a Mac. I've used this as well as the instruction on this repository and so far I'm getting the following error:

```HEAD is now at 5514e13f9 MIPSEL IRX changes
configure: error: Building GCC requires GMP 4.2+, MPFR 3.1.0+ and MPC 0.8.0+.
Try the --with-gmp, --with-mpfr and/or --with-mpc options to specify
their locations.  Source code for these libraries can be found at
their respective hosting sites as well as at
https://gcc.gnu.org/pub/gcc/infrastructure/.  See also
http://gcc.gnu.org/install/prerequisites.html for additional info.  If
you obtained GMP, MPFR and/or MPC from a vendor distribution package,
make sure that you have installed both the libraries and the header
files.  They may be located in separate packages.
../scripts/002-gcc-stage1.sh: Failed.
../scripts/002-iop.sh: Failed.
```

I'm sure this won't be the last problem I run into but if anybody with more experience could help I'm looking to learn more about every aspect involved with installing this SDK.

I don't know a ton about how terminals work. I followed a tutorial to configure iTerm2 but I honestly didn't understand a ton of it. I've put the following in ~/.zprofiles (presumably so it will run every time I open the terminal?) but I'm not sure if I've done that correctly

```export PS2DEV=/usr/local/ps2dev
export PATH=$PATH:$PS2DEV/bin
export PATH=$PATH:$PS2DEV/ee/bin
export PATH=$PATH:$PS2DEV/iop/bin
export PATH=$PATH:$PS2DEV/dvp/bin
export PS2SDK=$PS2DEV/ps2sdk
export PATH=$PATH:$PS2SDK/bin
```

Thanks in advance :)

**Comment by 7dog123** [comment, marked as answer]
It's best if you use https://github.com/ps2dev/ps2dev and follow [compilation.yml](https://github.com/ps2dev/ps2dev/blob/ba5b8c641a8cb7d5420a56ce2857d920be7d8668/.github/workflows/compilation.yml) Specifically Install macOS packages and Runs all the stages in the shell.

**Reply by jtgladden** [reply]
So I installed the packages as listed in compilation.yml and exported the gnu-sed path as listed and it got a lot farther but it still fails eventually...
```
../../gdb/python/py-gdb-readline.c:63:29: error: use of undeclared identifier '_PyOS_ReadlineTState'
      PyEval_RestoreThread (_PyOS_ReadlineTState);
                            ^
1 error generated.
make[2]: *** [python/py-gdb-readline.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [all-gdb] Error 2
make: *** [all] Error 2
../scripts/001-binutils.sh: Failed.
../scripts/003-ee.sh: Failed.
../scripts/001-toolchain.sh: Failed.
```

**Reply by 7dog123** [reply]
It's because `_PyOS_ReadlineTState` is no longer made  public in Python 3.13

**Reply by jtgladden** [reply]
So is there a work-around or can I just not build it?

**Reply by 7dog123** [reply]
Yeah downgrade back to python3.12

**Reply by jtgladden** [reply]
So it seems to have finished building but when I try and compile a sample I get the following error:

```❯ cd cpp-hello
❯ ls
Makefile     cpp-hello.cc
❯ make
mips64r5900el-ps2-elf-g++ -D_EE -G0 -O2 -Wall -gdwarf-2 -gz  -I/usr/local/ps2dev/ps2sdk/ee/include -I/usr/local/ps2dev/ps2sdk/common/include -I.  -c cpp-hello.cc -o cpp-hello.o
cpp-hello.cc:16:17: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
   16 | char * erl_id = "cpp-hello";
      |                 ^~~~~~~~~~~
cpp-hello.cc:18:5: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
   18 |     "libc",
      |     ^~~~~~
mips64r5900el-ps2-elf-gcc -mno-crt0 -O2 -o cpp-hello.erl cpp-hello.o -L/usr/local/ps2dev/ps2sdk/ee/lib -Wl,-zmax-page-size=128   -Wl,-r -Wl,-d
mips64r5900el-ps2-elf-gcc: error: unrecognized command-line option '-mno-crt0'; did you mean '-mno-crc'?
make: *** [cpp-hello.erl] Error 1
```

I read a few issues about  this [here](https://github.com/pspdev/psptoolchain/issues/78) but it's all going over my head. Any suggestions?

**Reply by 7dog123** [reply]
Sorry man. I don't know anything about that.

**Reply by AKuHAK** [reply]
@Jtgladden cpp sample is broken, you can see that it is commented here:
https://github.com/ps2dev/ps2sdk/blob/4c0214b73654d5dd493e96fc074f9dc8f2233551/samples/Makefile_sample#L18

if you want to test sample just type make sample within sdk directory
